scripts/test-gem-install

```diff
@@ -0,0 +1,33 @@
+#! /usr/bin/env bash
+#
+#  run as part of CI
+#
+if [[ $# -lt 1 ]] ; then
+  echo "usage: $(basename $0) <gems_dir> [install_flags]"
+  exit 1
+fi
+
+GEMS_DIR=$1
+shift
+INSTALL_FLAGS=$*
+
+test -e /etc/os-release && cat /etc/os-release
+
+set -e -x -u
+
+pushd $GEMS_DIR
+  gemfile=$(ls *.gem | head -n1)
+  ls -l ${gemfile}
+  gem install --no-document ${gemfile} -- ${INSTALL_FLAGS}
+  gem list -d re2
+popd
+
+if [ -n "${BUNDLE_APP_CONFIG:-}" ] ; then
+  export BUNDLE_CACHE_PATH="${BUNDLE_APP_CONFIG}/cache"
+fi
+
+bundle install --local || bundle install
+
+cd $(dirname `gem which re2`)
+cd ..
+rake spec
```

Comment:
I wonder if we could avoid including all the extra test files in the gem by running the test suite from the checkout but making sure the installed gem is first in the load path rather than the `lib` directory.

Reply:
Yes, that's definitely a possibility.

Reply:
I think we can use [`actions/checkout`'s `sparse-checkout`](https://github.com/actions/checkout#Fetch-only-the-root-files-and-github-and-src-folder) to do this: only checkout `spec` and `.rspec`, install the gem and appropriate version of RSpec and run `rspec` (avoiding the bundle and other development dependencies only relevant for the build).